File: templates/problems/15_greedy/03_jump_game_ii/jump_game_ii_test.go
package jump_game_ii

import "testing"

func TestJump(t *testing.T) {
	tests := []struct {
		name     string
		nums     []int
		expected int
	}{
		{"example 1", []int{2, 3, 1, 1, 4}, 2},
		{"example 2", []int{2, 3, 0, 1, 4}, 2},
		{"single element", []int{0}, 0},
		{"two elements", []int{1, 2}, 1},
		{"already at end", []int{1}, 0},
		{"linear jumps", []int{1, 1, 1, 1}, 3},
		{"large first jump", []int{5, 1, 1, 1, 1, 1}, 1},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got := jump(tt.nums)
			if got != tt.expected {
				t.Errorf("jump(%v) = %d, want %d", tt.nums, got, tt.expected)
			}
		})
	}
}

File: templates/problems/15_greedy/03_jump_game_ii/typescript/jumpGameIi.test.ts
import { describe, it, expect } from 'vitest';
import { jump } from './jumpGameIi';

describe('JumpGameII', () => {
    it('example 1', () => {
        expect(jump([2, 3, 1, 1, 4])).toBe(2);
    });

    it('example 2', () => {
        expect(jump([2, 3, 0, 1, 4])).toBe(2);
    });

    it('single element', () => {
        expect(jump([0])).toBe(0);
    });

    it('two elements', () => {
        expect(jump([1, 2])).toBe(1);
    });

    it('already at end', () => {
        expect(jump([1])).toBe(0);
    });

    it('linear jumps', () => {
        expect(jump([1, 1, 1, 1])).toBe(3);
    });

    it('large first jump', () => {
        expect(jump([5, 1, 1, 1, 1, 1])).toBe(1);
    });

    it('greedy per level', () => {
        expect(jump([1, 2, 1, 1, 1])).toBe(3);
    });

    it('level boundary', () => {
        expect(jump([7, 0, 9, 6, 9, 6, 1, 7, 9, 0, 1, 2, 9, 0, 3])).toBe(2);
    });
});
